### PR TITLE
Validate user input for cert-get-requestdata

### PR DESCRIPTION
**Pull request by Akasurde**
Fix adds validatation for Principal and CSR generation
tool values

Fixes https://pagure.io/freeipa/issue/6742

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

### CODE REVIEW

File: ipaclient/plugins/csrgen.py [hunk 1]
```diff
@@ -75,10 +75,19 @@ def execute(self, *args, **options):
             util.check_writable_file(options['out'])
 
         principal = options.get('principal')
+        if principal is None:
+            raise errors.InvocationError(message=u"Principal is required")
```
Comment (tiran) [inline ipaclient/plugins/csrgen.py]: Please include ```_()``` for i18n.

File: ipaclient/plugins/csrgen.py [hunk 2]
```diff
@@ -75,10 +75,19 @@ def execute(self, *args, **options):
...
         profile_id = options.get('profile_id')
         if profile_id is None:
             profile_id = dogtag.DEFAULT_PROFILE
         helper = options.get('helper')
+        if helper is None:
+            raise errors.InvocationError(message=u"CSR generation tool is"
```
Comment (tiran) [inline ipaclient/plugins/csrgen.py]: You are missing a trailing space in this line. You can make the code more readable by indenting the class' arguments:

```
            raise errors.InvocationError(
                message=_(u"CSR generation tool is required")
            )
```

File: ipaclient/plugins/csrgen.py [hunk 3]
```diff
@@ -75,10 +75,19 @@ def execute(self, *args, **options):
...
+                                                 "required")
+        if helper.lower() not in ['openssl', 'certutil']:
+            raise errors.InvocationError(message=u"Allowed values for CSR "
```
Comment (tiran) [inline ipaclient/plugins/csrgen.py]: same here

File: ipaclient/plugins/csrgen.py (hunk 1 above)
Comment (Akasurde) [inline ipaclient/plugins/csrgen.py]: OK

File: ipaclient/plugins/csrgen.py (hunk 2 above)
Comment (Akasurde) [inline ipaclient/plugins/csrgen.py]: OK.

File: ipaclient/plugins/csrgen.py (hunk 3 above)
Comment (Akasurde) [inline ipaclient/plugins/csrgen.py]: OK.

**Comment by rcritten** [comment]
You are duplicating the list of helpers. It would have been better to have helper defined as a StrEnum. If it isn't too late to change (e.g. no release has shipped with that in the API) then perhaps a separate patch, then you wouldn't need this enforcement at all.

**Comment by Akasurde** [comment]
@rcritten I don't know about backward compatibility of changing helper to StrEnum. @MartinBasti  @HonzaCholasta Can you please comment on this?

**Comment by MartinBasti** [comment]
I have no context about how exactly certrequest is supposed to work, but IMO it was done in that way to allow dynamically adding more helpers as plugins, that's why it is Str and not SrEnum, but code doesn't look it may support that.

@LiptonB do you remember why Str param was used?

@Akasurde Right now there is no backward compatibility because 4.5 will be first release that contains this feature.

**Comment by LiptonB** [comment]
I don't think one could really add a new helper without modifying the code, so there's probably no need to allow arbitrary strings. Given that, StrEnum seems appropriate.

For the record, https://github.com/freeipa/freeipa/pull/542 is going to be modified to remove the `helper` parameter of `cert-get-requestdata` entirely, though I haven't had a chance to make the change yet.

**Comment by MartinBasti** [comment]
> For the record, #542 removes the helper parameter of cert-get-requestdata, and will be modified to remove the concept of different helpers entirely, though I haven't had a chance to make that change yet.

today is 4.5 release so you have to keep some level of backward compatibility in that PR

**Comment by Akasurde** [comment]
@MartinBasti Should I wait for #542 to get merged?

**Comment by Akasurde** [comment]
Bump for review.

**Comment by tomaskrizek** [comment]
Closing for inactivity. If this PR is still relevant, please re-open it.

**Pull request closed without merging**